### PR TITLE
feat: upgrade to Angular 10

### PR DESCRIPTION
**Pull request by fabiomartino**
Hi,

With this PR the Angular dependency and file structure will be updated to latest (ATM of writing) to latest versions.

I reproduce the structure for the new json introduced in version 10, I mean tsconfig.base.json.

Still a warning (not due to this PR) with [angular-hmr](https://github.com/PatrickJS/angular-hmr)... it cause with the version 2.1.3
the common warning

```js
CommonJS or AMD dependencies can cause optimization bailouts.
```
CommonJS_or_AMD
I made a PR to "solve" this warning with latest angular version https://github.com/PatrickJS/angular-hmr/pull/87 

Tested with Angular + Material and Bootstrap as UI
Hope it help

Cordially

Fabio

### CODE REVIEW

**Comment by PatrickJS** [comment]
i published the hmr fixes as 3.0.0

**Comment by fabiomartino** [comment]
> i published the hmr fixes as 3.0.0

Thx [PatrickJS](https://github.com/PatrickJS)

**Review by sinedied — COMMENTED** [review]
Thanks a lot for this PR! 👍 

Looks good to me, except for my comment on polyfill could you take a look?

BTW, if you update `<preference name="android-minSdkVersion" value="19"/>` to `<preference name="android-minSdkVersion" value="22"/>` in __cordova._config.xml it should fix the CI for the mobile builds.

File: generators/app/templates/src/_polyfills.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/***************************************************************************************************
+ * Load `$localize` onto the global scope - used if i18n tags appear in Angular templates.
+ */
+import '@angular/localize/init';
```
[the remark this answers is not in the document]
Comment (fabiomartino) [reply generators/app/templates/src/_polyfills.ts:4]: In UI like Material also need the @angular/localize library as default for internationalizations...
but if you prefer we can restore the condition or maybe add also the material UI
```js
<% if (props.ui === 'bootstrap' || props.ui === 'material' ) { -%>
/***************************************************************************************************
 * Load `$localize` onto the global scope - used if i18n tags appear in Angular templates.
 */
import '@angular/localize/init';
<% } -%>
```

**Comment by fabiomartino** [comment]
> BTW, if you update `<preference name="android-minSdkVersion" value="19"/>` to `<preference name="android-minSdkVersion" value="22"/>` in __cordova._config.xml it should fix the CI for the mobile builds.

Change made.

**Review by fabiomartino — COMMENTED** [review]
> Looks good to me, except for my comment on polyfill could you take a look?

I made a comment... maybe restore the condition and add to it the material UI also?
let me know...

Cordially

Fabio

File: generators/app/templates/src/_polyfills.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sinedied) [reply generators/app/templates/src/_polyfills.ts:4]: Having the condition for bootstrap + material sounds good to me, as Ionic don't need it. Thanks!

[the remark this answers is not in the document]
Comment (fabiomartino) [reply generators/app/templates/src/_polyfills.ts:5]: Done!

**Review by sinedied — APPROVED** [review]

**Comment by sinedied** [comment]
Thanks! Expect a new release with your changes soon 🎉

**Pull request merged**